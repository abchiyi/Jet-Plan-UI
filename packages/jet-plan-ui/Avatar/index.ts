import type { App, Plugin } from "vue";
import Avatar from "./src/index.vue";

export const AvatarPlugin: Plugin = {
  install(app: App) {
    app.component("j-avatar", Avatar);
  },
};

export {
  Avatar,
};
