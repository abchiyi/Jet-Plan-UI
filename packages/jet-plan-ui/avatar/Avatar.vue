<template>
    <div v-auto-color class="j-avatar" :style="styles" :class="classes">
        <div class="content">
            <slot></slot>
        </div>
    </div>
</template>

<script>
import { AutoColor } from '../tool/directives';
export default {
    name: 'j-avatar',
    props: {
        border: {
            type: Boolean,
            default: false,
        },
        circle: {
            type: Boolean,
            default: true,
        },
        size: {
            type: [Number, String],
            default: '2',
        },
    },
    computed: {
        classes() {
            return [this.border ? 'border' : '', this.circle ? 'circle' : ''];
        },
        styles() {
            return {
                '--size': this.size + 'em',
            };
        },
    },
    directives: {
        AutoColor,
    },
};
</script>

<style >
.j-avatar {
    font-size: calc(var(--size) / 2);
    background: var(--white);
    min-height: var(--size);
    min-width: var(--size);
    display: inline-block;
    margin: 5px;
}

.j-avatar .content {
    justify-content: center;
    align-items: center;
    height: var(--size);
    display: flex;
}

.j-avatar.border {
    outline: 0.2em solid var(--border-dark);
}

.j-avatar img,
.j-avatar {
    border-radius: var(--m-radius);
}

.j-avatar.circle img,
.j-avatar.circle {
    border-radius: var(--size);
}

.j-avatar img {
    height: var(--size);
    width: var(--size);
}
</style>
