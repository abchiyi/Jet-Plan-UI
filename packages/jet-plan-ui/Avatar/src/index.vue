<template>
  <div :class="classes">
    <div class="content">
      <slot></slot>
    </div>
  </div>
</template>

<script lang="ts">
import { JET_THEME } from "jet-plan-ui/theme";
import { Bumper, autoTextColor } from "jet-plan-ui/tool";
import { defineComponent } from "vue";
export default defineComponent({
  name: "j-avatar",
  props: {
    circle: {
      type: Boolean,
      default: true,
    },
  },
  setup() {
    return {
      colorBackground: JET_THEME.Theme.background.light,
    };
  },
  data() {
    return {
      colorText: "",
      Bumper: new Bumper(20),
    };
  },
  mounted() {
    this.Bumper.setAlarmCooled(() => {
      this.colorText = autoTextColor(this.$el);
    });
    this.Bumper.action();
  },
  updated() {
    this.Bumper.action();
  },
  computed: {
    classes() {
      return ["j-avatar", this.circle ? "circle" : ""];
    },
  },
});
</script>

<style>
.j-avatar {
  background: v-bind("colorBackground");
  color: v-bind(colorText);
  display: inline-block;
  position: relative;
  font-size: 1.5rem;
  user-select: none;
  overflow: hidden;
  min-height: 2em;
  min-width: 2em;
  margin: 5px;
}

.j-avatar .content {
  justify-content: center;
  align-items: center;
  position: absolute;
  display: flex;
  bottom: 0;
  right: 0;
  left: 0;
  top: 0;
}

.j-avatar.circle img,
.j-avatar.circle {
  border-radius: 2em;
}

.j-avatar img,
.j-avatar {
  border-radius: var(--m-radius);
}

.j-avatar img {
  height: var(--size);
  width: var(--size);
}
</style>
