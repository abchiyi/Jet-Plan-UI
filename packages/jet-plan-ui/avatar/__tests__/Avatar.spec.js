import {
    mount
} from '@vue/test-utils'

import Avatar from '../Avatar.vue'

function cssAttrsStringToObj (str) {
    if (str) {
        const temp = {}
        const attrs = str.split(';')
        for (let n = 0; n < attrs.length; n++) {
            if (attrs[n].trim()) {
                let [key, value] = attrs[n].split(':')
                temp[key.trim()] = value.trim()
            }
        }
        return temp
    } else {
        return str
    }

}

describe('Avatar', () => {
    it("Document test", async () => {
        const wrapper = mount(Avatar)
        expect(wrapper.html()).toMatchSnapshot()

        await wrapper.setProps({
            border: true,
        })
        expect(wrapper.html()).toMatchSnapshot()

        await wrapper.setProps({
            border: true,
            circle: false,
        })
        expect(wrapper.html()).toMatchSnapshot()
        await wrapper.setProps({
            border: true,
            circle: false,
            color: '#61ff73'
        })
        expect(wrapper.html()).toMatchSnapshot()

    })

    it('Class', async () => {
        const wrapper = mount(Avatar)

        expect(wrapper.classes()).toContain(
            Avatar.name
        )

        expect(wrapper.classes()).toContain(
            'j-avatar'
        )
        expect(wrapper.classes()).toContain(
            'circle'
        )
        await wrapper.setProps({
            border: true
        })
        expect(wrapper.classes()).toContain(
            'border'
        )
        await wrapper.setProps({
            border: true,
            circle: false
        })
        expect(wrapper.classes()).not.toContain(
            'circle'
        )
    })

    it('Style value', async () => {
        const wrapper = mount(Avatar)

        const defaultSize = {
            size: '2em'
        }

        expect(
            cssAttrsStringToObj(
                wrapper.attributes('style')
            )['--size']
        ).toContain(defaultSize.size)

        await wrapper.setProps({
            size: 4
        })


        expect(
            cssAttrsStringToObj(
                wrapper.attributes('style')
            )['--size']
        ).toContain('4em')
    })
})
