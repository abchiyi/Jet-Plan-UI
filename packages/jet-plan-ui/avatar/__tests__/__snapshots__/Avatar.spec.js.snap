// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Document test 1`] = `
<div class="j-avatar circle" style="--size: 2em; transition: 1ms ease-in-out;">
  <div class="content"></div>
</div>
`;

exports[`Avatar Document test 2`] = `
<div class="j-avatar border circle" style="--size: 2em; transition: 1ms ease-in-out;">
  <div class="content"></div>
</div>
`;

exports[`Avatar Document test 3`] = `
<div class="j-avatar border" style="--size: 2em; transition: 1ms ease-in-out;">
  <div class="content"></div>
</div>
`;

exports[`Avatar Document test 4`] = `
<div class="j-avatar border" style="--size: 2em; transition: 1ms ease-in-out;" color="#61ff73">
  <div class="content"></div>
</div>
`;
